modules/assignment/src/main/java/com/intuit/wasabi/assignment/cache/impl/AssignmentsMetadataCacheImpl.java

```diff
@@ -0,0 +1,356 @@
+        Element val = cache.get(appName);
+        if(isNull(val)) {
+            expList = experimentRepository.getExperimentsForApps(singleEntrySet(appName)).get(appName);
+            cache.put(new Element(appName, expList));
+        } else {
+            expList = (List<Experiment>) val.getObjectValue();
+        }
+
+        return isNull(expList)?new ArrayList<>():expList;
+    }
+
+    /**
+     *
+     * @param expId
+     *
+     * @return An experiment for given experiment id.
+     */
+    @Override
+    public Optional<Experiment> getExperimentById(Experiment.ID expId) {
+        Experiment exp = null;
+        Cache cache = cacheManager.getCache(CACHE_NAME.EXPERIMENT_ID_TO_EXPERIMENT_CACHE.toString());
+        Element val = cache.get(expId);
+        if(isNull(val)) {
+            exp = experimentRepository.getExperimentsMap(singleEntrySet(expId)).get(expId);
+            cache.put(new Element(expId, exp));
+        } else {
+            exp = (Experiment) val.getObjectValue();
+        }
+
+        return Optional.ofNullable(exp);
+    }
+
+
+    /**
+     *
+     * @param appName
+     *
+     * @return prioritized list of experiments for given application.
+     */
+    @Override
+    public Optional<PrioritizedExperimentList> getPrioritizedExperimentListMap(Application.Name appName) {
+        PrioritizedExperimentList expList = null;
+        Cache cache = cacheManager.getCache(CACHE_NAME.APP_NAME_TO_PRIORITIZED_EXPERIMENTS_CACHE.toString());
+        Element val = cache.get(appName);
+        if(isNull(val)) {
+            expList = prioritiesRepository.getPriorities(singleEntrySet(appName)).get(appName);
+            cache.put(new Element(appName, expList));
+        } else {
+            expList = (PrioritizedExperimentList) val.getObjectValue();
+        }
+
+        return Optional.ofNullable(expList);
+    }
+
+    /**
+     *
+     * @param expId
+     *
+     * @return List of experiments which are mutually exclusive to the given experiment.
+     */
+    @Override
+    public List<Experiment.ID> getExclusionList(Experiment.ID expId) {
+        List<Experiment.ID> exclusionList = null;
+        Cache cache = cacheManager.getCache(CACHE_NAME.EXPERIMENT_ID_TO_EXCLUSION_CACHE.toString());
+        Element val = cache.get(expId);
+        if(isNull(val)) {
+            exclusionList = mutexRepository.getExclusivesList(singleEntrySet(expId)).get(expId);
+            cache.put(new Element(expId, exclusionList));
+        } else {
+            exclusionList = (List<Experiment.ID>) val.getObjectValue();
+        }
+
+        return isNull(exclusionList)?new ArrayList<>():exclusionList;
+    }
+
+    /**
+     *
+     * @param expId
+     * @return BucketList for given experiment.
+     */
+    @Override
+    public BucketList getBucketList(Experiment.ID expId) {
+        BucketList bucketList = null;
+        Cache cache = cacheManager.getCache(CACHE_NAME.EXPERIMENT_ID_TO_BUCKET_CACHE.toString());
```

Comment:
So if a experiment is updated, it will be refreshed for 5 minutes.  Will this cause problems if user is creating experiment with multiple buckets or is editing them ?